txdbus/test/client_tests.py

```diff
@@ -846,14 +846,16 @@ def on_proxy( ro ):
             ro.notifyOnSignal('sharedSignal', d2.callback,
                               interface='org.txdbus.trial.Signal2')
 
-            ro.callRemote('sendShared1')
-            ro.callRemote('sendShared2')
+            return defer.DeferredList([
+                ro.callRemote('sendShared1'),
+                ro.callRemote('sendShared2'),
+            ])
 
 
-        self.get_proxy().addCallback(on_proxy)
+        callRemotesDone = self.get_proxy().addCallback(on_proxy)
 
 
-        dl = defer.DeferredList( [d1,d2] )
+        dl = defer.DeferredList([d1,d2, callRemotesDone])
```

Comment:
Maybe I'm being slow here, but, shouldn't `d1` and `d2` return  strictly AFTER `callRemotesDone`, since the latter actually calls the former two?